pkg/util/sketch/series.go

```diff
@@ -0,0 +1,92 @@
+package sketch
+
+import (
+	"github.com/axiomhq/hyperloglog"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/grafana/loki/pkg/logproto"
+)
+
+const ValueTypeTopKMatrix = "topk_matrix"
+
+type TopKVector struct {
+	topk *Topk
+	ts   uint64
+}
+
+// TopkMatrix is `promql.Value` and `parser.Value`
+type TopKMatrix []TopKVector
+
+// Type implements `promql.Value` and `parser.Value`
+func (TopKMatrix) Type() parser.ValueType { return ValueTypeTopKMatrix }
+
+// String implements `promql.Value` and `parser.Value`
+func (TopKMatrix) String() string {
+	return ""
+}
+
+func (s TopKMatrix) ToProto() (*logproto.TopKMatrix, error) {
```

Comment:
Is there a way we can align the proto structure and the logqlmodel structure? That would let us avoid doing all this translation which would be faster & have less code to maintain.

Reply:
The original version did this. However, I didn't want the encoding to impact the performance of algorithm. Eg we might decide on a sparse matrix encoding that would not be fast for the access pattern of the algorithm. 

I'm open to change it back though.